song-server/src/main/java/org/icgc/dcc/song/server/service/CompositeEntityService.java

```diff
@@ -49,13 +49,14 @@ private static Sample buildPersistentSample(CompositeEntity s){
 
   public String save(String studyId, CompositeEntity s) {
     String id = sampleService.findByBusinessKey(studyId, s.getSampleSubmitterId());
-    s.setSampleId(id);
     if (isNull(id)) {
       val sampleCreateRequest = buildPersistentSample(s);
       s.setSpecimenId(getSampleParent(studyId, s));
       sampleCreateRequest.setSpecimenId(s.getSpecimenId());
       id = sampleService.create(studyId, sampleCreateRequest);
+      s.setSampleId(id);
```

Comment:
Is this line needed? `s` is never used after this as the method will immediately return `id` when it leaves the conditional.